pyvista/core/filters/data_set.py

```diff
@@ -3506,7 +3514,160 @@ def plot_over_line(
+        if figure:
+            plt.figure(figsize=figsize)
+        # Plot it in 2D
+        if values.ndim > 1:
+            for i in range(values.shape[1]):
+                plt.plot(distance, values[:, i], label=f'Component {i}')
+            plt.legend()
+        else:
+            plt.plot(distance, values)
+        plt.xlabel('Distance')
+        if ylabel is None:
+            plt.ylabel(scalars)
+        else:
+            plt.ylabel(ylabel)
+        if title is None:
+            plt.title(f'{scalars} Profile')
+        else:
+            plt.title(title)
+        if fname:
+            plt.savefig(fname)
+        if show:  # pragma: no cover
+            plt.show()
+
+    def sample_over_broken_line(self, points=None, tolerance=None, progress_bar=False):
+        """Sample a dataset onto a broken line.
+
+        Parameters
+        ----------
+        points : np.ndarray or list, optional
+            List of points defining a broken line, default is ``None``
+            If given, pointa, pointb and resolution will be ignored.
+
+        tolerance : float, optional
+            Tolerance used to compute whether a point in the source is in a
+            cell of the input.  If not given, tolerance is automatically generated.
+
+        progress_bar : bool, optional
+            Display a progress bar to indicate progress.
+
+        Returns
+        -------
+        pyvista.PolyData
+            Line object with sampled data from dataset.
+
+        Examples
+        --------
+        Sample over a plane that is interpolating a point cloud.
+
+        >>> import pyvista
+        >>> import numpy as np
+        >>> np.random.seed(12)
+        >>> point_cloud = np.random.random((5, 3))
+        >>> point_cloud[:, 2] = 0
+        >>> point_cloud -= point_cloud.mean(0)
+        >>> pdata = pyvista.PolyData(point_cloud)
+        >>> pdata['values'] = np.random.random(5)
+        >>> plane = pyvista.Plane()
+        >>> plane.clear_data()
+        >>> plane = plane.interpolate(pdata, sharpness=3.5)
+        >>> sample = plane.sample_over_broken_line([[-0.5, -0.5, 0], [0.5, -0.5, 0], [0.5, 0.5, 0]])
+        >>> pl = pyvista.Plotter()
+        >>> _ = pl.add_mesh(pdata, render_points_as_spheres=True, point_size=50)
+        >>> _ = pl.add_mesh(sample, scalars='values', line_width=10)
+        >>> _ = pl.add_mesh(plane, scalars='values', style='wireframe')
+        >>> pl.show()
+
+        """
+        # Make a broken line and sample the dataset
+        broken_line = pyvista.Line(points=points)
+        sampled_broken_line = broken_line.sample(
+            self, tolerance=tolerance, progress_bar=progress_bar
+        )
+        return sampled_broken_line
+
+    def plot_over_broken_line(
+        self,
+        points=None,
+        scalars=None,
+        title=None,
+        ylabel=None,
+        figsize=None,
+        figure=True,
+        show=True,
+        tolerance=None,
+        fname=None,
+        progress_bar=False,
+    ):
```

Comment:
I'm not sure how many more "plot ... " functions that plot with matplotlib we should add...

Reply:
Personally, I don't need matplotlib plotting as long as I have the "sample" function to do it myself. As you say, a large number of plot functions is redundant for a library, so I will not add them.

Reply:
The only reason I'd say to add it is if *you* need it. Otherwise, let's leave it out